sop-website/docs/Waveform-Data/Waveform-Extension.mdx

```diff
@@ -0,0 +1,222 @@
+| 3     | procedure_occurrence_id       | int       | Conditionally Required | Populate if the waveform is tied to a documented clinical procedure (e.g., diagnostic ECG, EEG study). Extract from EHR or metadata tags; if not available, leave null.                         |
+| 4     | device_exposure_id            | int       | Optional               | Link to device record if available (e.g., from ICU device logs or telemetry registry). If a device used is known (Philips monitor, EEG cap), map via ETL joins; else leave null.                |
+| 5     | waveform_channel_source_value | string    | Recommended            | Use channel label from the raw waveform file (e.g., “Lead II”, “ECG I”, “SpO2”, “ABP”). If not present, derive from channel index or use placeholder (“Channel 1”).                             |
+| 6     | channel_concept_id            | int       | Yes                    | Map the channel label or signal type to a standard OMOP concept (2-billion range or community extension). Use a lookup table for common physiological signals. Log unmapped entries for review. |
+| 7     | metadata_source_value         | string    | Yes                    | Populate with the metadata type, such as “sampling_rate”, “gain”, “calibration_factor”, “compression_ratio”. Extracted from header fields or external metadata.                                 |
+| 8     | metadata_concept_id           | int       | Yes                    | Map the metadata_source_value to a standard OMOP concept (e.g., "Sampling rate" → CONCEPT_ID = X). Maintain an internal vocabulary map; flag unknowns.                                          |
+| 9     | value_as_string               | string    | Optional               | Use for non-numeric, human-readable metadata (e.g., “DC coupling”, “auto-scaled”, “2x compression”). Store the raw metadata value as a string if it does not fit numeric or concept fields.     |
+| 10    | value_as_number               | float     | Optional               | Use if the metadata is numeric (e.g., sampling_rate = 500, gain = 0.2). Validate precision and units. Use float type.                                                                           |
+| 11    | value_as_concept_id           | int       | Optional               | Use if the value is categorical and can be mapped to an OMOP concept (e.g., “Invasive” → concept ID, “High Quality” → concept ID). Optional if stored in value_as_string.                       |
+| 12    | unit_concept_id               | int       | Recommended            | Populate for physical values (e.g., Hz, mmHg, mV) using OMOP standard units. Use unit lookup table or join against raw units found in header.                                                   |
```

Comment:
Consider ordering the `value_as` fields in this order: `value_as_number`, `value_as_concept_id`, `value_as_string`. This puts string at the end which seems to make sense since it's the fallback option. It also makes this order consistent with the Waveform Feature table.